From 4dbb4dc5bce6617824c38b2adc251b1add27d017 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Mon, 7 Nov 2016 11:10:12 +0100
Subject: [PATCH] Use versioned Obsoletes

Also change <= N to < N+1. For example, dracut-029-1 > dracut-029, so would not
get obsoleted properly. This all applies to old versions, so doesn't make much
difference in practice, so just fix it to avoid c&p duplication of the bad
pattern in the future.
---
 dracut.spec | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/dracut.spec b/dracut.spec
index 7b9d675..0515006 100644
--- a/dracut.spec
+++ b/dracut.spec
@@ -33,7 +33,7 @@ License: GPLv2+ and LGPLv2+
 URL: https://dracut.wiki.kernel.org/
 
 # Source can be generated by
-# http://git.kernel.org/?p=boot/dracut/dracut.git;a=snapshot;h=%{version};sf=tgz
+# http://git.kernel.org/?p=boot/dracut/dracut.git;a=snapshot;h=%%{version};sf=tgz
 Source0: http://www.kernel.org/pub/linux/utils/boot/dracut/dracut-%{version}.tar.xz
 Source1: https://www.gnu.org/licenses/lgpl-2.1.txt
 
@@ -68,10 +68,10 @@ BuildRequires: asciidoc
 %if 0%{?fedora} > 12 || 0%{?rhel}
 # no "provides", because dracut does not offer
 # all functionality of the obsoleted packages
-Obsoletes: mkinitrd <= 6.0.93
-Obsoletes: mkinitrd-devel <= 6.0.93
-Obsoletes: nash <= 6.0.93
-Obsoletes: libbdevid-python <= 6.0.93
+Obsoletes: mkinitrd < 6.0.94
+Obsoletes: mkinitrd-devel < 6.0.94
+Obsoletes: nash < 6.0.94
+Obsoletes: libbdevid-python < 6.0.94
 %endif
 
 %if 0%{?fedora} > 16 || 0%{?rhel} > 6
@@ -86,9 +86,9 @@ Provides: mkinitrd = 2.6.1
 Obsoletes: dracut-kernel < 005
 Provides:  dracut-kernel = %{version}-%{release}
 
-Obsoletes: dracut <= 029
-Obsoletes: dracut-norescue
-Provides:  dracut-norescue
+Obsoletes: dracut < 030
+Obsoletes: dracut-norescue < 030
+Provides:  dracut-norescue = %{version}-%{release}
 
 Requires: bash >= 4
 Requires: coreutils
@@ -195,8 +195,8 @@ initramfs with dracut, with live image capabilities, like Live CDs.
 %package config-generic
 Summary: dracut configuration to turn off hostonly image generation
 Requires: %{name} = %{version}-%{release}
-Obsoletes: dracut-nohostonly
-Provides:  dracut-nohostonly
+Obsoletes: dracut-nohostonly < 030
+Provides:  dracut-nohostonly = %{version}-%{release}
 
 %description config-generic
 This package provides the configuration to turn off the host specific initramfs
@@ -205,7 +205,7 @@ generation with dracut and generates a generic image by default.
 %package config-rescue
 Summary: dracut configuration to turn on rescue image generation
 Requires: %{name} = %{version}-%{release}
-Obsoletes: dracut <= 029
+Obsoletes: dracut < 030
 
 %description config-rescue
 This package provides the configuration to turn on the rescue initramfs
